cli/src/test/java/org/jboss/pnc/bacon/cli/AppTest.java

```diff
@@ -76,90 +85,88 @@ void testJSONDisabled() throws Exception {
-                                        "--profile",
-                                        "foobar",
-                                        "admin",
-                                        "maintenance-mode",
-                                        "activate",
-                                        "-h" })));
-
-        assertEquals(
-                "Usage: bacon pnc admin maintenance-mode activate [-hov]\n"
-                        + "       [-p=<configurationFileLocation>] [--profile=<profile>] <reason>\n"
-                        + "This will disable any new builds from being accepted\n"
-                        + "      <reason>              Reason\n"
-                        + "  -h, --help                display this help message\n"
-                        + "  -o                        use json for output (default to yaml)\n"
-                        + "  -p, --configPath=<configurationFileLocation>\n"
-                        + "                            Path to PNC configuration folder\n"
-                        + "      --profile=<profile>   PNC Configuration profile\n"
-                        + "  -v, --verbose             Verbose output\n" + "\n" + "Example:\n"
-                        + "$ bacon pnc admin maintenance-mode activate \"Switching to maintenance mode for\n"
-                        + "upcoming migration\"\n",
-                text);
-
-        text = tapSystemErr(
-                () -> assertEquals(
-                        0,
-                        app.run(
-                                new String[] {
-                                        "--configPath",
-                                        configYaml.toString(),
-                                        "--verbose",
-                                        "pnc",
-                                        "--profile",
-                                        "foobar",
-                                        "admin",
-                                        "maintenance-mode",
-                                        "activate",
-                                        "-h" })));
-        assertEquals(StringUtils.countMatches(text, "Config file set from flag with profile foobar to"), 1);
+        restoreSystemProperties(() -> {
+            System.setProperty("picocli.ansi", "false");
+            App app = new App();
+            String text = tapSystemOut(
+                    () -> assertThat(
+                            app.run(
+                                    new String[] {
+                                            "--verbose",
+                                            "pnc",
+                                            "--profile",
+                                            "foobar",
+                                            "admin",
+                                            "maintenance-mode",
+                                            "activate",
+                                            "-h" })).isZero());
```

Comment:
Incorrect use of hardcoded line separators here instead of what is actually used, `System.lineSeparator()`. It is also possible to strip them by replacing `tapSystemOut()` with `tapSystemOutNormalized()`.

Reply:
This passed on ci though....

Reply:
Perhaps you mean only to run your tests in one particular environment then:

```@EnabledIfSystemProperty(named = "line.separator", matches = "\\n")```

or

```
@EnabledOnOs(value = {OS.LINUX})
```

Reply:
The `EnabledOnOs` is not so good as what determines the correctness of the test is not at all what OS it is on, but what the code under test, in this case picocli, outputs. If picocli outputs `\n`, then hardcoding `\n` in the test would be 100% correct. If picocli outputs `System.lineSeparator()`, then you want to test for `System.lineSeparator()`. The point is to match the expected output of picocli in this case.

Reply:
That will do `return tapSystemOut(statement).replace(lineSeparator(), "\n");` This is a minimal one-word change then instead of changing every line in that test.

Reply:
That sounds better then....